Update docs

jnnr opened the pull request:
The docs do not reflect the current state of the repo. To make it easier to get started with the model, this PR provides an update.

https://oemof-flexmex.readthedocs.io/en/features-update-docs/

jnnr committed:
Remove outdated information and link to docs

jnnr committed:
Adapt docs and add TODOs

jnnr committed:
Adapt and add TODOs to model pipeline

jnnr committed:
Correct information on input data

jnnr committed:
Correct link

jnnr committed:
Merge branch 'dev' into features/update-docs

jnnr committed:
Move section on extra parameters to model structure

monika-o commented on docs/overview.rst:
```diff
@@ -7,13 +7,18 @@ Overview
 oemof-flexmex is sector-integrated multi-node energy system model featuring a lot of flexibility options.
```
missing article ("a")

monika-o commented on docs/overview.rst:
```diff
@@ -7,13 +7,18 @@ Overview
 Its region, interconnections and components can be adapted flexibly.
 
+The model has been developed in the context of the model comparison project FlexMex and builds upon the open energy modeling framework _oemof_.
+oemof-flexmex defines a data format to run energy system optimization using oemof.solph and oemof.tabular. It provides pre- and postprocessing routines
```
A person who is not familiar with oemof might get confused by the abundance of the term here. Maybe insert a short explanation on what oemof and oemof.solph are? Which other energy models are there that are built on oemof and what are the characteristics od oemof-flexmex compared to them? Aren't flexibility options relevant in all energy models? Perhaps this is beyond the scope of an overview - one could consider explaining it in a separate section. Anyway, I would insert links to the respective github-sites here.

jnnr replied:
I agree - there should be a sentence on what oemof is and that it houses several libraries for different purposes together with a link to the oemof.org-Website and the links to oemof.solph and oemof.tabular.

monika-o commented on docs/model_pipeline.rst:
```diff
@@ -117,6 +121,7 @@ The found filenames are interpreted according to the following pattern::
 .. note:: ``Experiment name`` and ``year`` are ignored at the moment.
```
I'm wondering how to implement different time series for different scenarios if the experiment name is ignored here. What exactly does it mean?
In general, a section on scenarios would be helpful. How are scenarios defined? In which format do they need to be written? 'expandable' and 'greenfield' is False by default, right? What is the 'scenario select' section made for? How do I  link scalars and how do I link time series to a scenario, and how do scenarios (e.g. FlexMex1, FlexMex2) and sub-scenarios (FlexMex1a, FlexMex2_1b ..) differ in their implementation? What does 'scenario_overwrite' do?

monika-o commented on docs/model_structure.rst:
```diff
@@ -30,12 +31,24 @@ The filenames for the components are of the form carrier-tech.csv (e.g. :file:`e
 * **carrier** Energy sector according to carrier (e.g. solar, wind, biomass, coal, lignite, uranium, oil, gas, methane, hydro, waste, electricity, heat).
 * **tech** Specification of the technology (e.g. st, ocgt, ccgt, pv, onshore, offshore, ror, phs, extchp, bpchp, battery)
```
I don't understand all the abbreviations (and I don't know if other people do). If they are not already in there, I would add demand, curtailment, shortage and transmission or at least some of these.

monika-o commented on docs/model_structure.rst:
```diff
@@ -30,12 +31,24 @@ The filenames for the components are of the form carrier-tech.csv (e.g. :file:`e
 Sequences
 ---------
 
 The filenames are of the form type_profile (e.g.
```
Isn't the form rather carrier-tech_profile?

jnnr replied:
Yes. Resolved with 5c07463

monika-o commented on docs/model_structure.rst:
```diff
@@ -30,12 +31,24 @@ The filenames for the components are of the form carrier-tech.csv (e.g. :file:`e
+
 Available components
 ====================
 
```
In my opinion it would be clearer without the middle column, by stating the path once for all. I don't know if it's possible in an rst file to remove a column.

monika-o reviewed (commented):
Here are the thoughts and questions I had when reading through the documentation. I will probably add to that later on while using the model.

jnnr committed:
Update overview

jnnr committed:
Fix typo

jnnr committed:
Extend list of techs

jnnr committed:
Correct info on filename structure

monika-o committed:
Fix typo

monika-o commented on docs/getting_started.rst:
```diff
@@ -63,7 +63,7 @@ Required data
 * Raw input data, see :ref:`input data format`.
```
I am not sure whether I have built the docs correctly, but in my version these links don't work.

monika-o commented on docs/getting_started.rst:
```diff
@@ -63,7 +63,7 @@ Required data
 * Output template data, see :ref:`postprocessing`.
 
-This data will be provided on github in the future.
+This data is planned to be published at a later point in time by the project FlexMex.
```
```suggestion
An example of this data is planned to be published at a later point in time by the project FlexMex.
```

monika-o replied:
I propose this change because the former sentence suggests that the model was not usable before the data is published.

monika-o replied:
In the docs for the model pipeline it says "Raw data for FlexMex is not part of the oemof-flexmex github repository but can be provided by the FlexMex project partners." I would make clearer at both places when and where people can obtain the data.

monika-o committed:
Add autosectionlabel extension
This extension is needed so that references by section title work.

monika-o committed:
Activate references

monika-o committed:
Convert TODOs for snakemake into (very short) paragraph

monika-o committed:
Shorten overview

In my opinion, the last sentence doubles partly the previous one and is
too detailed for an overview.

monika-o committed:
Improve readability

monika-o committed:
Change section order
It's easier to follow when the system is explained before explaining how
to vary it in the scenarios.

monika-o committed:
Explain how to use solver other than cbc.

monika-o committed:
Rename "scalars" into "parameters"
This is easier to understand. The model terminology is explained in the
model pipeline.

monika-o committed:
Add example for timeseries input

monika-o committed:
correct typo

monika-o committed:
Improve readability a little

monika-o committed:
Change synthax for file paths

monika-o committed:
Resolve some TODO's and create new ones

monika-o committed:
Add link to regions section

monika-o committed:
Use preprocessed file for component description
In the previous commits, I had described the component files from the
model structure instead of those from the preprocessed data. This is now
corrected.

monika-o committed:
Remove superfluous TODOs

monika-o committed:
Make clear that oemof-flexmex isn't further developed

monika-o committed:
Some minor changes

monika-o committed:
Restructure overview

monika-o committed:
Merge branch 'dev' into features/update-docs

monika-o committed:
Apply black

monika-o committed:
Use more recent black version
I hope that this will resolve the ImportError "cannot import name
'_unicodefun' from 'click'".

monika-o committed:
Fix versions for oemof and oemof.tabular
